### PR TITLE
Lock icon flash

### PR DESCRIPTION
**Pull request by 0tuedon**
This PR closes the first half of  #674 

Lock Icon no longer flashes

### CODE REVIEW

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 1]
```diff
@@ -101,18 +104,27 @@ export default function Tab({
...
-        )}
+        {!(currentIndex >= index) &&
+          !isCompleted &&
+          pnLessonId.split('-')[0] !== 'outro' && (
+            <Icon
+              icon="lock"
+              className="absolute right-[10px] top-[10px] h-3 w-3 opacity-50"
+            />
+          )}
+        {currentIndex > index ||
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:115]:
```suggestion
        {(currentIndex > index) ||
```

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 2]
```diff
@@ -101,18 +104,27 @@ export default function Tab({
...
+          (isCompleted && (
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:116]:
```suggestion
        {(currentIndex > index) || isCompleted && (
```

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 3]
```diff
@@ -101,18 +104,26 @@ export default function Tab({
...
+              icon="lock"
+              className="absolute right-[10px] top-[10px] h-3 w-3 opacity-50"
+            />
+          )}
+        {(currentIndex > index) || isCompleted && (
+            <Icon
+              icon="check"
+              className="absolute right-[5px] top-[5px] h-[20px] w-[20px]"
+            />
+          )) ||
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:120]:
```suggestion
        ) ||
```

**Comment by benalleng** [comment]
the behavior where the previous challenge isn't showing a check has returned...

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 4]
```diff
@@ -62,13 +62,16 @@ export default function Tab({
...
   const isLast = index === count - 1
   const lessonHref =
     challenge.lessonId === chapters[slug].metadata.challenges[0]
       ? chapters[slug].metadata.intros[0]
       : challenge.lessonId
   const href = `${routes.chaptersUrl}/${slug}/${lessonHref}`
-
+  const currentIndex = chapters[slug].metadata.challenges.indexOf(
+    challengeId + '-1'
+  )
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:74]:
```suggestion
  const currentIndex = chapters[slug].metadata.challenges/indexOf(
    challengeId + '-1'
  )
  const challengeLock = (currentIndex < index) && !isCompleted && (pnLessonId.split('-')[0] !== 'outro')
  const challengeCheck = (isCompleted || (currentIndex > index)) || (pnLessonId.split('-')[0] === 'outro')
```

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 5]
```diff
@@ -101,18 +104,26 @@ export default function Tab({
...
+              icon="check"
+              className="absolute right-[5px] top-[5px] h-[20px] w-[20px]"
+            />
+          ) ||
+          (pnLessonId.split('-')[0] === 'outro' && (
+            <Icon
+              icon="check"
+              className="absolute right-[5px] top-[5px] h-[20px] w-[20px]"
+            />
+          ))}
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:126]:
```suggestion
        {challengeLock && (
          <Icon
            icon="lock"
            className="absolute right-[10px] top-[10px] h-3 w-3 opacity-50"
          />
        )}
        {challengeCheck && (
          <Icon
            icon="check"
            className="absolute right-[5px] top-[5px] h-[20px] w-[20px]"
          />
        )}
```

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 6]
```diff
@@ -62,13 +62,16 @@ export default function Tab({
         .replace('intro', chapters[slug].metadata.challenges[0].split('-')[0])
     : undefined
   const isActive = challenge.lessonId.split('-')[0] === challengeId
+  // const currentIndex =
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:65]:
```suggestion
```

File: components/Navbar/NavbarDesktop/Tab.tsx [hunk 7]
```diff
@@ -68,7 +68,11 @@ export default function Tab({
       ? chapters[slug].metadata.intros[0]
       : challenge.lessonId
   const href = `${routes.chaptersUrl}/${slug}/${lessonHref}`
-
+  const currentIndex = chapters[slug].metadata.challenges/indexOf(
```
Comment (benalleng) [inline components/Navbar/NavbarDesktop/Tab.tsx:71]:
```suggestion
  const currentIndex = chapters[slug].metadata.challenges.indexOf(
```

**Comment by benalleng** [comment]
I was able to fix all the errors except the fact that the flash still occurs when coming back through the chapter a second time. I need to think a little more about this lgic to see if its even possible to prevent that behavior since the lock is now only considering progress and not the index on the second pass

**Review by daviroo — APPROVED** [review]

**Pull request merged**